src/Extensions/TimedScopes/TimedScopeProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Logging.TimedScopes
+{
+	internal class TimedScopeProvider : ITimedScopeProvider
+	{
+		public TimedScopeProvider(IMachineInformation machineInformation, TimedScopeEventSource eventSource)
+		{
+			m_serviceName = machineInformation.ServiceName;
+			m_eventSource = eventSource;
+		}
+
+		public TimedScope Start(string name, TimedScopeResult result) =>
+			new TimedScope(m_eventSource, m_serviceName, name, result);
+
+		private readonly TimedScopeEventSource m_eventSource;
+		private readonly string m_serviceName;
+	}
+
+
+	/// <summary>Interface to create TimedScope</summary>
+	public interface ITimedScopeProvider
+	{
+		/// <summary>Create and start TimedScope</summary>
+		TimedScope Start(string name, TimedScopeResult result = TimedScopeResult.SystemError);
+	}
+
+
+	/// <summary>Class to log duration of activity</summary>
+	public class TimedScope : IDisposable
+	{
+		/// <summary>TimedScope result</summary>
+		public TimedScopeResult Result { get; set; }
+
+
+		/// <summary>TimedScope sub tipe</summary>
+		public string SubType { get; set; }
+
+
+		/// <summary>TimedScope meta data</summary>
+		public string MetaData { get; set; }
```

Comment:
>MetaData [](start = 16, length = 8)

Nitpick: `Metadata` is one word so doesn't require a capital `D`

Reply:
Copied from previous implementation, but I agree it makes sense to fix, thanks